### ISSUE TITLE
Deletes unexpected folders starting with the given name

### ISSUE THREAD

**Issue by ernest-ns** [s3tools/s3cmd#470]
I was executing the following command
```s3cmd del --recursive s3://<bucket>/app/public/spree/artwork_preview```

But, strangely s3cmd deleted both ```/app/public/spree/artwork_preview``` and ```/app/public/spree/artwork_preview_thumbnail```. 

This seems very dangerous. Any idea why it is happening?

**Comment by mdomsch** [comment]
I agree that's unexpected.  It's doing a pattern match on the string you
gave it (remember, s3cmd doesn't really know about directories), and is
deleting everything that matches the string.  If you append a '/' as the
last character of your delete command, it works as expected, because the /
is in the name of objects below that point.

In general, '/' characters change the behaviour of the program exactly
because of this (similar to how rsync works: see man rsync for details).
